File: solutions/2006/main.py
from collections import defaultdict
from typing import List 

class Solution:
    def countKDifference(self, nums: List[int], k: int) -> int:
        result = 0 
        nums_dict = defaultdict(int)
        for n in nums: 
            if n in nums_dict:
                result += nums_dict[n]
            nums_dict[n+k] += 1 
            nums_dict[n-k] += 1 
        return result 

if __name__ == "__main__": 
    nums = [1,2,2,1]
    k = 1 
    s = Solution()
    assert s.countKDifference(nums, k) == 4

File: solutions/2006/main.rs
use std::collections::HashMap; 

struct Solution {}

impl Solution {
    pub fn count_k_difference(nums: Vec<i32>, k: i32) -> i32 {
        let mut umap = HashMap::new(); 
        let mut result = 0; 
        for n in nums {
            result += *umap.entry(n).or_insert(0); 
            *umap.entry(n-k).or_insert(0) += 1; 
            *umap.entry(n+k).or_insert(0) += 1; 
        }
        result 
    }
}

fn main() {
    assert_eq!(Solution::count_k_difference(vec![1,2,2,1], 1), 4); 
}